File: data/code_pairs/fortran/869.f90
module BoxMullerMod
    implicit none
contains
    subroutine boxMuller(x, y)
        real, intent(out) :: x, y
        real :: u1, u2, r, theta
        call random_seed()
        call random_number(u1)
        call random_number(u2)
        r = sqrt(-2.0 * log(u1))
        theta = 2.0 * 3.14159265358979323846 * u2
        x = r * cos(theta)
        y = r * sin(theta)
    end subroutine boxMuller
end module BoxMullerMod

File: data/code_pairs/cpp/869.cpp
#include <cmath>
#include <utility>
#include <random>

std::pair<double, double> boxMuller() {
    static std::random_device rd;
    static std::mt19937 gen(rd());
    static std::uniform_real_distribution<> dis(0.0, 1.0);

    double u1 = dis(gen);
    double u2 = dis(gen);
    double r = sqrt(-2.0 * log(u1));
    double theta = 2.0 * M_PI * u2;
    double x = r * cos(theta);
    double y = r * sin(theta);

    return std::make_pair(x, y);
}